### PR TITLE
arch: set (*running_task)->xcp.regs to NULL when exit from irq/exception

### PR DESCRIPTION
**Pull request by hujun260**
## Summary

arch: set (*running_task)->xcp.regs to NULL when exit from irq/exception
reason:
(*running_task)->xcp.regs is invalid when in threadcontext, we marke it as NULL to avoid misusage

## Impact
arch

## Testing
qemu-armv7a:smp

### CODE REVIEW

File: arch/xtensa/src/common/xtensa_irqdispatch.c [hunk 1]
```diff
@@ -114,5 +114,13 @@ uint32_t *xtensa_irq_dispatch(int irq, uint32_t *regs)
...
   board_autoled_off(LED_INIRQ);
+
+  regs = tcb->xcp.regs;
+
+  /* (*running_task)->xcp.regs is about to become invalid
+   * and will be marked as NULL to avoid misusage.
+   */
+
+  (*running_task)->xcp.regs = NULL;
   return tcb->xcp.regs;
```
Comment (xiaoxiang781216) [inline arch/xtensa/src/common/xtensa_irqdispatch.c:125]:
```suggestion
  return regs;
```

File: arch/risc-v/src/common/supervisor/riscv_perform_syscall.c [hunk 2]
```diff
@@ -89,5 +89,10 @@ void *riscv_perform_syscall(uintreg_t *regs)
 
   up_set_interrupt_context(false);
 
+  /* (*running_task)->xcp.regs is about to become invalid
+   * and will be marked as NULL to avoid misusage.
+   */
+
+  (*running_task)->xcp.regs = NULL;
   return tcb->xcp.regs;
```
Comment (xiaoxiang781216) [inline arch/risc-v/src/common/supervisor/riscv_perform_syscall.c:97]: do we need save regs beore zero

File: arch/risc-v/src/common/riscv_doirq.c [hunk 3]
```diff
@@ -138,5 +138,11 @@ uintreg_t *riscv_doirq(int irq, uintreg_t *regs)
 
 #endif
   board_autoled_off(LED_INIRQ);
+
+  /* (*running_task)->xcp.regs is about to become invalid
+   * and will be marked as NULL to avoid misusage.
+   */
+
+  (*running_task)->xcp.regs = NULL;
   return tcb->xcp.regs;
```
Comment (xiaoxiang781216) [inline arch/risc-v/src/common/riscv_doirq.c:147]: ditto

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



No. This PR description does not meet the NuttX requirements.  Here's why:

* **Insufficient Summary:** The summary needs more detail.  While it states *what* is changed, it doesn't adequately explain *why* this change is necessary. What problems did the previous behavior cause? What specific "misusage" are you preventing? What are the benefits of this change?
* **Missing Impact Details:** While "arch" is mentioned, the impact section needs to be much more specific. *Which* architectures are affected? Are *all* architectures affected? Will any particular boards or drivers need modification?  Explicitly answer the YES/NO questions and provide descriptions.  For example:
    * Impact on hardware (will arch(s) / board(s) / driver(s) change)? YES (This change affects all ARMv7-A architectures. No board or driver modifications are expected.)
* **Insufficient Testing Information:**  "qemu-armv7a:smp" is not enough. Provide the full build host details (OS, CPU, compiler version). Also,  the testing logs sections are empty.  You *must* include logs demonstrating the issue before the change and the improved behavior after the change.  Simply stating that it "works as intended" is not acceptable.  Show how you tested and what the results were.
* **Missing Issue References:** Are there any related NuttX issues that prompted this change?  If so, those should be linked.


**In short, the PR needs to provide more context, be more specific about its impact, and include actual testing logs to demonstrate the fix.**

File: arch/risc-v/src/common/supervisor/riscv_perform_syscall.c (hunk 2 above)
Comment (hujun260) [reply arch/risc-v/src/common/supervisor/riscv_perform_syscall.c]: done

File: arch/tricore/src/common/tricore_doirq.c [hunk 4]
```diff
@@ -117,6 +117,11 @@ IFX_INTERRUPT_INTERNAL(tricore_doirq, 0, 255)
 
   up_set_current_regs(NULL);
```
Comment (xiaoxiang781216) [inline arch/tricore/src/common/tricore_doirq.c:118]: let's remove up_set_current_regs/up_get_current_regs in tricore

**Review by xiaoxiang781216 — APPROVED** [review]

File: arch/tricore/src/common/tricore_doirq.c (hunk 4 above)
Comment (hujun260) [reply arch/tricore/src/common/tricore_doirq.c:118]: ok
